uaa/src/main/webapp/WEB-INF/spring/saml-providers.xml

```diff
@@ -247,17 +256,27 @@
         </bean>
 
         <!-- Bindings, encoders and decoders used for creating and parsing messages -->
-        <bean id="postBinding" class="org.springframework.security.saml.processor.HTTPPostBinding">
+        <bean id="postBinding" class="org.cloudfoundry.identity.uaa.provider.saml.UaaHTTPPostBinding">
             <constructor-arg ref="parserPool" />
-            <constructor-arg ref="velocityEngine" />
+            <constructor-arg ref="httpPostSimpleSignEncoder"/>
         </bean>
 
         <bean id="assertionBinding" class="org.cloudfoundry.identity.uaa.authentication.SamlAssertionBinding">
             <constructor-arg ref="parserPool" />
         </bean>
 
-        <bean id="redirectBinding" class="org.springframework.security.saml.processor.HTTPRedirectDeflateBinding">
+        <bean id="redirectBinding" class="org.cloudfoundry.identity.uaa.provider.saml.UaaHTTPRedirectDeflateBinding">
             <constructor-arg ref="parserPool" />
+            <constructor-arg ref="httpRedirectDeflateEncoder"/>
+        </bean>
+
+        <bean id ="httpRedirectDeflateEncoder" class="org.cloudfoundry.identity.uaa.provider.saml.UaaHTTPRedirectDeflateEncoder">
+            <property name="securityConfiguration" ref="zoneAwareSamlSecurityConfiguration"/>
+        </bean>
+
+        <bean id ="httpPostSimpleSignEncoder" class="org.cloudfoundry.identity.uaa.provider.saml.UaaHTTPPostSimpleSignEncoder">
```

Comment:
From what I'm seeing, the true zonification are in  the custom encoders, and the other uaa subclasses are just to inject this encoder. Could we avoid having to override the post and redirect bindings by creating new decoders in xml and injecting them into the springframework binding beans?

Reply:
I agree, this can be refactored.